File: Cpp/57_Insert_Interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> result;
        vector<int> interval = newInterval;

        for(int i = 0; i < intervals.size(); ++i){
            if(interval[0] > intervals[i][1]){
                result.push_back(intervals[i]);
            } else if(interval[1] < intervals[i][0]){
                result.push_back(interval);
                interval = intervals[i];
            } else if(interval[0] <= intervals[i][1] || interval[1] >= intervals[i][0]){
                interval[0] = min(interval[0], intervals[i][0]);
                interval[1] = max(interval[1], intervals[i][1]);
            }
        }
        result.push_back(interval);

        return result;
    }
};

File: Kotlin/57_Insert_Interval.kt
class Solution {
    fun insert(intervals: Array<IntArray>, newInterval: IntArray): Array<IntArray> {
        val result = ArrayList<IntArray>()
        var temp = newInterval

        for(interval in intervals){
            if(temp[0] > interval[1]){
                result.add(interval)
            } else if(temp[1] < interval[0]){
                result.add(temp)
                temp = interval
            } else if(temp[0] <= interval[1] || temp[1] >= interval[0]){
                temp[0] = minOf(temp[0], interval[0])
                temp[1] = maxOf(temp[1], interval[1])
            }
        }
        result.add(temp);

        return result.toTypedArray()
    }
}